File: src/test/java/cz/vhromada/catalog/web/mapper/SongMapperTest.java
package cz.vhromada.catalog.web.mapper;

import static org.assertj.core.api.Assertions.assertThat;

import cz.vhromada.catalog.entity.Song;
import cz.vhromada.catalog.web.common.SongUtils;
import cz.vhromada.catalog.web.fo.SongFO;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.mapstruct.factory.Mappers;

/**
 * A class represents test for mapper between {@link Song} and {@link SongFO}.
 *
 * @author Vladimir Hromada
 */
class SongMapperTest {

    /**
     * Mapper for songs
     */
    private SongMapper mapper;

    /**
     * Initializes mapper.
     */
    @BeforeEach
    void setUp() {
        mapper = Mappers.getMapper(SongMapper.class);
    }

    /**
     * Test method for {@link SongMapper#map(Song)}.
     */
    @Test
    void map() {
        final Song song = SongUtils.getSong();

        final SongFO songFO = mapper.map(song);

        SongUtils.assertSongDeepEquals(songFO, song);
    }

    /**
     * Test method for {@link SongMapper#map(Song)} with null song.
     */
    @Test
    void map_NullSong() {
        assertThat(mapper.map(null)).isNull();
    }

    /**
     * Test method for {@link SongMapper#mapBack(SongFO)}.
     */
    @Test
    void mapBack() {
        final SongFO songFO = SongUtils.getSongFO();

        final Song song = mapper.mapBack(songFO);

        SongUtils.assertSongDeepEquals(songFO, song);
    }

    /**
     * Test method for {@link SongMapper#mapBack(SongFO)} with null song.
     */
    @Test
    void mapBack_NullSong() {
        assertThat(mapper.mapBack(null)).isNull();
    }

}

File: src/test/kotlin/cz/vhromada/catalog/web/mapper/SongMapperTest.kt
package cz.vhromada.catalog.web.mapper

import cz.vhromada.catalog.entity.Song
import cz.vhromada.catalog.web.CatalogMapperTestConfiguration
import cz.vhromada.catalog.web.common.SongUtils
import cz.vhromada.catalog.web.fo.SongFO
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.test.context.ContextConfiguration
import org.springframework.test.context.junit.jupiter.SpringExtension

/**
 * A class represents test for mapper between [Song] and [SongFO].
 *
 * @author Vladimir Hromada
 */
@ExtendWith(SpringExtension::class)
@ContextConfiguration(classes = [CatalogMapperTestConfiguration::class])
class SongMapperTest {

    /**
     * Mapper for songs
     */
    @Autowired
    private lateinit var mapper: SongMapper

    /**
     * Test method for [SongMapper.map].
     */
    @Test
    fun map() {
        val song = SongUtils.getSong()

        val songFO = mapper.map(song)

        SongUtils.assertSongDeepEquals(songFO, song)
    }

    /**
     * Test method for [SongMapper.mapBack].
     */
    @Test
    fun mapBack() {
        val songFO = SongUtils.getSongFO()

        val song = mapper.mapBack(songFO)

        SongUtils.assertSongDeepEquals(songFO, song)
    }

}
